src/cli/setup_mode.rs

```diff
@@ -25,12 +25,12 @@ use crate::{
     before_help = format!("rustup-init {}", common::version()),
 )]
 struct RustupInit {
-    /// Enable verbose output
-    #[arg(short, long)]
+    /// Enable verbose output, limit console logger level to 'DEBUG' if 'RUSTUP_LOG' is unset
```

Comment:
Conceptually it feels like CLI flags should have higher specificity than environment variables. I think the reason not to do so here is because the environment variable is more expressive (in the sense that it can set levels by crate etc), right?

The docstring seems a little verbose/awkward, maybe just "Set log level to 'DEBUG' (if 'RUSTUP_LOG' is unset)"?

Why use `overrides_with` here instead of `conflicts_with`? The meaning of `overrides_with` is not clear to me in case both are specified -- maybe it means last wins?

Reply:
> Conceptually it feels like CLI flags should have higher specificity than environment variables. I think the reason not to do so here is because the environment variable is more expressive (in the sense that it can set levels by crate etc), right?

Yes, I'd like to say that this is `RUSTUP_LOG` instead of something like [`CARGO_TERM_VERBOSE`](https://doc.rust-lang.org/cargo/reference/config.html#termverbose).

> The docstring seems a little verbose/awkward, maybe just "Set log level to 'DEBUG' (if 'RUSTUP_LOG' is unset)"?

Will fix.

> Why use `overrides_with` here instead of `conflicts_with`? The meaning of `overrides_with` is not clear to me in case both are specified -- maybe it means last wins?

Yes, this was stated in the PR description. However looking at what `cargo` currently does with `--quiet` and `--verbose`, I guess it'll still be better to revert this change...